src/Mantasflowers.Services/Services/User/UserService.cs

```diff
@@ -83,26 +85,95 @@ public async Task<string> GetUserUidByGuidAsync(Guid id)
             return user?.Uid;
         }
 
-        public async Task<PostCreateUserResponse> CreateUserAsync(string uid)
+        public async Task<PostCreateUserResponse> CreateUserAsync(string email, string password)
         {
+            var firebaseUser = await _fbService.CreateUserAsync(email, password);
+
+            if (string.IsNullOrWhiteSpace(firebaseUser.Uid))
+            {
+                throw new FirebaseUidNotFoundException("Firebase did not return Uid after creating user");
+            }
+
             var user = new Domain.Entities.User
             {
-                Uid = uid
+                Uid = firebaseUser.Uid
             };
 
+            await _unitOfWork.UserRepository.CreateAsync(user);
+            
             try
             {
-                await _unitOfWork.UserRepository.CreateAsync(user);
                 await _unitOfWork.SaveChangesAsync();
             }
             catch (DbUpdateException)
             {
+                await _fbService.DeleteUserByUidAsync(firebaseUser.Uid);
                 throw new FailedToAddDatabaseResourceException("Failed to create user");
             }
 
-            var resposne = _mapper.Map<PostCreateUserResponse>(user);
+            var resposne = _mapper.Map<PostCreateUserResponse>(user,
+                o => o.AfterMap((source, destination) =>
+                {
+                    destination.LoginEmail = firebaseUser.Email;
+                }));
 
             return resposne;
         }
+
+        public async Task DeleteUserAsync(string uid)
+        {
+            var user = await _unitOfWork.UserRepository.GetUserGraphByUidAsync(uid);
+
+            if (user == null)
+            {
+                throw new FirebaseUidNotFoundException($"No database user matching the {nameof(uid)} could be found");
+            }
+
+            await _fbService.DeleteUserByUidAsync(uid);
+
+            // These entities aren't linked for automatic cascading. Need to delete by hand.
+            var ordersToDelete = user.UserOrders.Select(x => x.Order);
+            var shipmentsToDelete = ordersToDelete.Select(x => x.Shipment);
+            var paymentsToDelete = ordersToDelete.Select(x => x.Payment);
```

Comment:
Already submitted Orders should never be deleted.

Reply:
Why not? If we don't delete them when deleting the user, they will essentially be there forever, with no purpose to serve. Also GDPR pepe be agri.

Reply:
Having a hole in a order history would be hard to explain, when paying taxes against fraud. Also, order does not contain any information on a user. You'd have to delete from UserOrder table, though.

Reply:
Makes sense. Will adjust. UserOrder already contains cascading.